buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy

```diff
@@ -38,6 +40,8 @@ public class GenerateMeasurementsTask extends DefaultTask {
 
         assert project.tasks.withType(JacocoReport).size() == 1 : 'Found multiple tasks which generate coverage reports.'
         coverageTaskName = project.tasks.withType(JacocoReport)[0].name
+
+        dependsOn(findAllProductCoverageTasks())
```

Comment:
When exactly does this run again? Where is this task created?

Reply:
This task is used in `tools/measurement/coverage/coverage.gradle`.

I'm not sure about the exact time. But at the time when the gradle file is evaluated, this task is already configured to depend on what `findAllProductCoverageTasks` returns.

Reply:
It can be verified by appending `generateCoverageMeasurements.dependsOn.forEach { println it }` to `coverage.gradle` and run `./gradlew`.

Reply:
Interesting. The Jacoco plugin is applied at configuration time, but you moved this up to evaluation time. It must be something to do with the subproject layout.